### PR TITLE
Docs prep work

### PR DESCRIPTION
**Pull request by cicdw**
I'm currently preparing to incorporate some major simplifying changes to our content, but before I update the content itself I wanted to do a high level pass and touch up a few sections.

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ base = ""
 command = """
 pip install --upgrade pip \
 && pip install --upgrade --upgrade-strategy eager  -e ".[dev]" \
-&& pip install git+https://oauth:${MKDOCS_MATERIAL_INSIDERS_REPO_RO}@github.com/PrefectHQ/mkdocs-material-insiders.git@master \
+&& pip install git+https://oauth:${MKDOCS_MATERIAL_INSIDERS_REPO_RO}@github.com/PrefectHQ/mkdocs-material-insiders.git@a1c87bd61619694c6b2375108afc5d072fbe4b07 \
```
Comment (cicdw) [inline netlify.toml:13]: this pin is necessary due to a python 3.8 incompatible type annotation in mkdocs insiders; it appears to have been fixed upstream but we haven't incorporated the fix yet - I'll remove the pin once the relevant update has been merged into our fork

**Review by jlowin — APPROVED** [review]

File: docs/index.md [hunk 2]
```diff
@@ -33,195 +33,13 @@ With Prefect you gain:
...
-```
-</div>
-
-
-You'll find a link directing you to the flow run page conveniently positioned at the top of your flow logs.
-
-![Alt text](img/ui/flow-run-diagram.jpg)
-
-Local flow run execution is great for development and testing, but to [schedule flow runs](/concepts/schedules/) or [trigger them based on events](/cloud/automations/), you’ll need to [deploy](/concepts/deployments/) your flows.
+Get up and running quickly with the [quickstart guide](getting-started/quickstart) and [tutorial](/tutorial/); for deeper dives on specific use cases, explore our [guides](guides/index/) for common scenarios.  Take your understanding even further with Prefect's [concepts](/concepts/index/) and [API reference](/api-ref/).  Join Prefect's [vibrant community](/community/) of over 26,000 engineers to learn with others and share your knowledge!
```
Comment (discdiver) [inline docs/index.md:40]:
```suggestion
Get up and running quickly with the [quickstart guide](/getting-started/quickstart/). 
Want more hands on practice to productionize your workflows? Follow our [tutorial](/tutorial/). 
For deeper dives on common use cases, explore our [guides](/guides/index/). 
Take your understanding even further with Prefect's [concepts](/concepts/index/) and [API reference](/api-ref/). 
Join Prefect's [vibrant community](/community/) of over 26,000 engineers to learn with others and share your knowledge!
```

Comment (discdiver) [reply docs/index.md:40]: Formatted links for consistency. 

Would like to try to differentiate when to use the quick start vs. tutorial, so added the language above.

**Review by discdiver — COMMENTED** [review]
Cool. I'm down with the quickstart as a standalone page and API tweaks.

File: mkdocs.yml [hunk 3]
```diff
@@ -179,12 +180,6 @@ nav:
           - Task Runners: concepts/task-runners.md
           - Events: concepts/events.md
           - Automations: concepts/automations.md
-          - Filesystems: concepts/filesystems.md
-          - Block and Agent-Based Deployments:
-                - Deployments: concepts/deployments-block-based.md
-                - Infrastructure: concepts/infrastructure.md
-                - Storage: concepts/storage.md
-                - Agents: concepts/agents.md
```
Comment (EmilRex) [inline mkdocs.yml:187]: @cicdw @discdiver I think it's important to preserve access to these concepts. What do y'all think about having a _note_ or _callout_ on both the deployments and workers page that points to these docs? This way they are not featured prominently in the nav, but still very visible.

Comment (discdiver) [reply mkdocs.yml:187]: Agree we want to make sure users can find them. We added them back to the nav in #10528 @EmilRex.

**Pull request merged**